sources/bloodhound/src/bin/kubernetes-checks/checks.rs

```diff
@@ -0,0 +1,724 @@
+            rotate_kubelet_server_certificate: bool,
+        }
+
+        #[derive(Deserialize)]
+        struct KubeletConfig {
+            #[serde(rename = "featureGates")]
+            feature_gates: FeatureGates,
+        }
+
+        let mut result = CheckerResult::default();
+
+        if let Ok(kubelet_file) = File::open(KUBELET_CONF_FILE) {
+            if let Ok(config) = serde_yaml::from_reader::<_, KubeletConfig>(kubelet_file) {
+                if !config.feature_gates.rotate_kubelet_server_certificate {
+                    result.error = "Kubelet RotateKubeletServerCertificate is disabled".to_string();
+                    result.status = CheckStatus::FAIL;
+                } else {
+                    result.status = CheckStatus::PASS;
+                }
+            } else {
+                // Feature gate has been defaulted to enabled since k8s 1.12, so if it is not found that is fine
+                result.status = CheckStatus::PASS;
+            }
+        } else {
+            result.error = format!("unable to read '{}'", KUBELET_CONF_FILE);
+        }
+
+        result
+    }
+
+    fn metadata(&self) -> CheckerMetadata {
+        CheckerMetadata {
+            title: "Verify that the RotateKubeletServerCertificate argument is set to true"
+                .to_string(),
+            id: "4.2.11".to_string(),
+            level: 1,
+            name: "k8s04021100".to_string(),
+            mode: Mode::Automatic,
+        }
+    }
+}
+
+// =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<=
+
+pub struct K8S04021200Checker {}
+
+impl Checker for K8S04021200Checker {
+    fn execute(&self) -> CheckerResult {
+        let allowed_suites: HashSet<&str> = vec![
+            "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
+            "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+            "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305",
+            "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+            "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305",
+            "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
+            "TLS_RSA_WITH_AES_256_GCM_SHA384",
+            "TLS_RSA_WITH_AES_128_GCM_SHA256",
+        ]
+        .into_iter()
+        .collect();
+
+        #[derive(Deserialize)]
+        struct KubeletConfig {
+            #[serde(rename = "tlsCipherSuites")]
+            tls_cipher_suites: Vec<String>,
+        }
+
+        let mut result = CheckerResult::default();
+
+        if let Ok(kubelet_file) = File::open(KUBELET_CONF_FILE) {
+            if let Ok(config) = serde_yaml::from_reader::<_, KubeletConfig>(kubelet_file) {
+                let configured_suites: HashSet<&str> = config
+                    .tls_cipher_suites
+                    .iter()
+                    .map(|s| s.as_str())
+                    .collect();
+                if !configured_suites.is_subset(&allowed_suites) {
+                    result.error = "Found disallowed cipher suites".to_string();
```

Comment:
Should the culprits be included in this error?

Reply:
This is another case where we would want them to refer to the benchmark to evaluate things. This would need to be something they are doing in a custom build, since we know the official Bottlerocket variants do not include any cipher suites that are not acceptable.